kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/LogReplaySuite.scala

```diff
@@ -295,7 +295,8 @@ class LogReplaySuite extends AnyFunSuite with TestUtils {
     assert(snapshot.isInstanceOf[SnapshotImpl])
     val snapshotImpl = snapshot.asInstanceOf[SnapshotImpl]
 
-    assert(snapshotImpl.getLatestTransactionVersion("fakeAppId") === Optional.of(3L))
-    assert(snapshotImpl.getLatestTransactionVersion("nonExistentAppId") === Optional.empty())
+    assert(snapshotImpl.getLatestTransactionVersion("fakeAppId", defaultEngine) === Optional.of(3L))
+    assert(snapshotImpl.getLatestTransactionVersion(
```

Comment:
Does `snapshotImpl.getLatestTransactionVersion("nonExistentAppId", defaultEngine).isEmpty)` work here?

Reply:
Works.

Reply:
It seems that Optional#isEmpty() is a Java 11 method, which is a shortcut for Java 8's !Optional#isPresent(). I think we should use the latter one for compatibility?